src/core/store/modules/vaults/vaults.actions.ts

```diff
@@ -174,30 +176,62 @@ const getExpectedTransactionOutcome = createAsyncThunk<
 /*                             Transaction Methods                            */
 /* -------------------------------------------------------------------------- */
 
-const approveDeposit = createAsyncThunk<void, { vaultAddress: string; tokenAddress: string }, ThunkAPI>(
+const approveDeposit = createAsyncThunk<
+  { amount: string; spenderAddress: string },
+  { vaultAddress: string; tokenAddress: string },
+  ThunkAPI
+>(
   'vaults/approveDeposit',
-  async ({ vaultAddress, tokenAddress }, { dispatch, getState }) => {
-    const { vaults } = getState();
+  async ({ vaultAddress, tokenAddress }, { dispatch, getState, extra }) => {
+    const { vaults, wallet, network } = getState();
     const vaultData = vaults.vaultsMap[vaultAddress];
-    const isZapin = vaultData.tokenId !== tokenAddress;
-    const spenderAddress = isZapin ? getConfig().CONTRACT_ADDRESSES.zapIn : vaultAddress;
+    const { vaultService, transactionService } = extra.services;
+    const amount = extra.config.MAX_UINT256;
+
+    const accountAddress = wallet.selectedAddress;
+    const isZapin = vaultData.token !== tokenAddress;
+    if (!accountAddress) throw new Error('WALLET NOT CONNECTED');
+
+    const tx = await vaultService.approveDeposit({
+      network: network.current,
+      accountAddress,
+      tokenAddress,
+      amount,
+      vaultAddress: vaultData.address,
+    });
+
+    if (typeof tx !== 'boolean') {
+      await transactionService.handleTransaction({ tx: tx as TransactionResponse, network: network.current });
+    }
 
-    const result = await dispatch(TokensActions.approve({ tokenAddress, spenderAddress }));
-    unwrapResult(result);
+    return { amount, spenderAddress: isZapin ? getConfig().CONTRACT_ADDRESSES.zapIn : vaultAddress };
```

Comment:
Since spender can be also partner tracking contract, would be best to dispatch `getVaultAllowance` after transaction is approved, so that state gets updated correctly ?

Reply:
fixed on b69e516c61c60f31501bd291946417d970f1b2ad